### PR TITLE
Add SLE15 Dockerfile and run our tests on it

### PR DESCRIPTION
**Pull request by lagartoflojo**
We were not testing Connect on SLES15. This PR changes that.

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/93475/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-build-image.
( Failure: Merged 5d893925f4871149ea0eaa82858c7b4044785a71 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/93476/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests.
( Failure: Merged d7a20dbfbfa39f038b89ba8e15e2aa3eceadca34 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/93756/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle15sp0.
( Failure: Merged 789099f51cb4fc60dcb84a84b51b91ee8168a9ae into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/94897/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp0.
( Failure: Merged d5fec488a29eb77a11432330c805bc278fa44243 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/94898/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp0.
( Failure: Merged 4a4fece8609874146041287b46d5e3d17cd8012f into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/94956/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp0.
( Failure: Merged 8fae2f91e1a2a488e4bce3b6bdf4dfe8b956e2da into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/94961/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp0.
( Failure: Merged d5e0a7d56b39be0f0ece5f38a006a295c9d7e64b into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/94967/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp0.
( Failure: Merged 72b063130728370cda1ce4a362904babc5f10aae into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/94973/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 94396c8a955f2ba799a3d02a8f9862b0f4f23cef into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/SCC-Connect-Prophet/94974/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project connect on step docker-run-tests-sle12sp2.
( Failure: Merged 3c457882840d4596833af4629256870f40383039 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by lagartoflojo** [comment]
I'll check on that failing spec, but it seems there's something wrong with the repo, not with our code.

```
command 'zypper --non-interactive --releasever 12.2 ref -f' failed
  Scenario: Rollback can be called on activated system without an issue         # features/rollback.feature:13
    When I call the migration rollback method                                   # features/step_definition/library_steps.rb:11
      Signature verification failed for file 'repomd.xml' from repository 'SLES12-SP2-Updates'.
      Repository 'SLES12-SP2-Updates' is invalid.
      [SUSE_Linux_Enterprise_Server_12_SP2_x86_64:SLES12-SP2-Updates|https://updates.suse.com/SUSE/Updates/SLE-SERVER/12-SP2/x86_64/update?fqoKLTWkyVVCsgtI3XUXo7tFmCVZ2TXj_kmIG3DdKLE0aSdqpyBsyH_DxKhmDr_E6rO-ShEpYrpmPAjlBQbPJICUouWqFJ58pTsYnJ_D0huLqEih2U510eZL0jsArUOzXE0Dus8mxlHbIvo] Valid metadata not found at specified URL
      Please check if the URIs defined for this repository are pointing to a valid repository.
      Skipping repository 'SLES12-SP2-Updates' because of the above error.
      Some of the repositories have not been refreshed because of an error. (SUSE::Connect::ZypperError)
      ./lib/suse/toolkit/system_calls.rb:22:in `execute'
      ./lib/suse/connect/zypper.rb:198:in `call'
      ./lib/suse/connect/zypper.rb:165:in `set_release_version'
      ./lib/suse/connect/migration.rb:42:in `rollback'
      ./features/step_definition/library_steps.rb:12:in `/^I call the migration rollback method$/'
      features/rollback.feature:14:in `When I call the migration rollback method'
```

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/95160/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 3c457882840d4596833af4629256870f40383039 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

File: Dockerfile.15sp0 [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM registry.scc.suse.de/suse/sles15:ga-beta7
```
Comment (ikapelyukhin) [inline Dockerfile.15sp0]: Maybe we should set the tag `sles15:ga` here, and in the registry tag the most recent image with this tag.

File: Gemfile [hunk 2]
```diff
@@ -22,7 +22,7 @@ group :test do
   gem 'rspec', '~> 3.3.0'
   gem 'rspec-its'
   gem 'webmock', '~> 1.21.0'
-  gem 'http', '~> 0.8.12'
-  gem 'aruba', '~> 0.7.4'
-  gem 'cucumber', '~> 2.4.0'
+  gem 'http', '~> 2.2.2'
+  gem 'aruba', '~> 0.14.3'
```
Comment (ikapelyukhin) [inline Gemfile]: Let's update to `0.14.4` and drop the `absolute_paths.rb` file.

Comment (lagartoflojo) [reply Gemfile]: Done :D

Comment (ikapelyukhin) [reply Gemfile]: You were faster! :grimacing:

File: Dockerfile.15sp0 (hunk 1 above)
Comment (lagartoflojo) [reply Dockerfile.15sp0]: Yeah, I wasn't too sure about that.
On one hand, I like the explicitness of `sles15:ga-beta7` -- that's the version we are currently testing. The downside is that we'll have to update the dockerfile as soon as RC1, RC2,..., GA are released.
On the other hand, using `sles15:ga` we can leave the dockerfile unchanged forever, but we don't *really* know if the image is a beta/rc or actually GA (until GA gets released).
`docker-ls` shows:
```
> docker-ls tags suse/sles15 --registry http://registry.scc.suse.de
requesting list . done
repository: suse/sles15
tags:
- ga
- ga-beta6
- ga-beta7
- sp0
```

TBH I kind of wish we didn't have those `sp0` and `ga` tags UNTIL those products became official... Right now, I have no idea on which beta `ga` and `sp0` got tagged, so I can't rely on them.

Comment (ikapelyukhin) [reply Dockerfile.15sp0]: Well, it's not too hard to tell which image it is:

```
$ docker-ls tag suse/sles15:ga --registry http://registry.scc.suse.de
requesting manifest . done
repository: suse/sles15
tagName: ga
digest: sha256:a04db4ae11a6499c9ad407adbf7d84030f45640b789842628b9e0b0b98e9be51
layers:
- sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
- sha256:e285d47f6e1f15e96218a62151440a8519d39e433d5ea2f20611cd1a7f17b3bf
```

```
$ docker-ls tag suse/sles15:ga-beta7 --registry http://registry.scc.suse.de
requesting manifest . done
repository: suse/sles15
tagName: ga-beta7
digest: sha256:4064d0802957330de7f800d80cd484025d5a3b373c7a7a8425a82f57d1b3edcf
layers:
- sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4
- sha256:e285d47f6e1f15e96218a62151440a8519d39e433d5ea2f20611cd1a7f17b3bf
```

That's the same image, which makes sense. :grimacing:

Comment (lagartoflojo) [reply Dockerfile.15sp0]: Could have been, could not have been. Depends on who tagged them and when. It's not so cool having to look at layer digests to check if they're the same image.. that's the whole point of the tag I guess.
Basically my problem is that the tag is `ga` but the image is not really GA, it's beta 7.

Comment (ikapelyukhin) [reply Dockerfile.15sp0]: Updating the tag in the file is also fine, just something we need not to forget to do.

Comment (lagartoflojo) [reply Dockerfile.15sp0]: Yup :)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/95202/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged 091dbb8e5a6729af9bf20013498753d4482111c5 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/SCC-Connect-Prophet/95203/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
( Success: Merged ad8b719ab097f908b71a0fa3ea43ddf266301d65 into 78a78f7f4a172a6285ab32f25cdc25d27d6cd6a9 )

**Pull request merged**